pkg/controller/istiocontrolplane/istiocontrolplane_controllerv2.go

```diff
@@ -0,0 +1,97 @@
+package istiocontrolplane
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+	"istio.io/operator/pkg/helmreconciler"
+)
+
+// ReconcileIstioControlPlane reconciles a v1alpha2 IstioControlPlane object
+type ReconcileIstioControlPlaneV2 struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client   client.Client
+	scheme   *runtime.Scheme
+	factory  *helmreconciler.Factory
+	instance runtime.Object
+}
+
+// Reconcile reads that state of the cluster for a v1alpha2 IstioControlPlane object and makes changes based on the state read
+// and what is in the IstioControlPlane.Spec
+func (r *ReconcileIstioControlPlaneV2) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling IstioControlPlane")
+
+	// Fetch the IstioControlPlane instance
+	instance := &v1alpha2.IstioControlPlane{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	deleted := instance.GetDeletionTimestamp() != nil
+	finalizers := instance.GetFinalizers()
+	finalizerIndex := indexOf(finalizers, finalizer)
+
+	if deleted {
+		if finalizerIndex < 0 {
+			reqLogger.Info("IstioControlPlane deleted")
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Info("Deleting IstioControlPlane")
+
+		reconciler, err := r.factory.NewISCPReconciler(instance, r.client, reqLogger)
+		if err == nil {
+			err = reconciler.Delete()
+		} else {
+			reqLogger.Error(err, "failed to create reconciler")
+		}
+		// XXX: for now, nuke the resources, regardless of errors
```

Comment:
what does XXX represent?  A TODO?  If so, please change it as such.

Reply:
Done.